projects/gloo/pkg/plugins/kubernetes/uds_convert.go

```diff
@@ -20,6 +21,39 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
+// these labels are used to propagate internal data
+// on synthetic Gloo resources generated from other Kubernetes
+// resources (generally Service).
+// The `~` is an invalid character that prevents these labels from ending up
+// on actual Kubernetes resources.
+const (
+	// KubeSourceResourceLabel indicates the kind of resource that the synthetic
+	// resource is based on.
+	KubeSourceResourceLabel = "~internal.solo.io/kubernetes-source-resource"
```

Comment:
getting errors like: `"caller":"discovery/discovery.go:151","msg":"failed reconciling upstreams","version":"1.0.0-ci1","discovered_by":"kubernetesplugin","upstreams":14,"error":"reconciling resource kube-system-kube-dns-53: creating kube resource kube-system-kube-dns-53: Upstream.gloo.solo.io \"kube-system-kube-dns-53\" is invalid: [metadata.labels: Invalid value: \"~internal.solo.io/kubernetes-name\": prefix part a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*'),` when running this in a cluster